src/PubMed/pubmed_sql_utils.jl

```diff
@@ -220,118 +392,152 @@ function get_article_mesh_by_concept(db, pmid::Integer, umls_concepts...; query_
+
+            path = joinpath(csv_path, "$(csv_prefix)$(table).csv")
+            cols_string = assemble_cols(df)
+
+            # Save article data (MySQL.stream from df)
+            ins_sql = """LOAD DATA LOCAL INFILE '$path' INTO TABLE $table CHARACTER SET latin1 FIELDS TERMINATED BY ',' OPTIONALLY ENCLOSED BY '"' IGNORE 1 LINES ($cols_string)"""
+            MySQL.execute!(db, ins_sql)
         end
     end
+
+    if cleanup
+        remove_csvs(articles, csv_path, csv_prefix)
+    end
+
+    return nothing
+
 end
 
-function db_insert!(db, pmid::Int64, mesh_heading_list::MeshHeadingList, verbose=false)
-    
-    for heading in mesh_heading_list
-        did_int = heading.descriptor_id
-        descriptor_name = heading.descriptor_name
-        dmjr = heading.descriptor_mjr
-
-
-        #Save Descriptor
-        insert_row!(db, "mesh_descriptor",
-        Dict(:id=>did_int,
-             :name=>descriptor_name),
-             verbose)
-
-        if isempty(heading.qualifier_id)
-            #Save Headings
-            insert_row!(db, "mesh_heading",
-            Dict(:id=>missing,
-                 :pmid=> pmid,
-                 :did=>did_int,
-                 :qid=>missing,
-                 :dmjr=>missing, :qmjr=>missing), verbose )
-        else
-
-            for i=1:length(heading.qualifier_id)
-                qid_int = heading.qualifier_id[i]
-                qualifier_name = heading.qualifier_name[i]
-                qmjr = heading.qualifier_mjr[i]
-
-                #Save Qualifiers`
-                insert_row!(db, "mesh_qualifier",
-                Dict(:id=>qid_int,
-                     :name=>qualifier_name),
-                     verbose )
-
-                #Save Headings
-                insert_row!(db, "mesh_heading",
-                Dict(:id=>missing,
-                     :pmid=> pmid,
-                     :did=>did_int,
-                     :qid=>qid_int,
-                     :dmjr=>dmjr, :qmjr=>qmjr), verbose )
+function db_insert!(db::SQLite.DB, articles::Dict{String,DataFrame}, csv_path::String = pwd(), csv_prefix::String = "$(Date(now()))_PubMed_"; verbose=false, cleanup=false)
+
+    #Insert csv prefix into files_meta talbe
+    meta_sql = """INSERT INTO file_meta (file_name,ins_start_time) VALUES ('$csv_prefix',CURRENT_TIMESTAMP)"""
+    SQLite.execute!(db, meta_sql)
+
+    for (table, df) in articles
+
+        # check if column names all exist in mysql table
+        if !col_match(db, table, df)
+            error("each DataFrame column must match the name of a table column")
+        end
+
+        for i = 1:size(df)[1]
+            col_dict = Dict{Symbol,Any}()
+            for col in df.colindex.names
+                col_dict[col] = df[i,col]
             end
+            insert_row!(db, table, col_dict)
         end
 
     end
+
+    meta_sql = """UPDATE file_meta SET ins_end_time = CURRENT_TIMESTAMP WHERE file_name = '$csv_prefix'"""
+    SQLite.execute!(db, meta_sql)
+
+    return nothing
+
 end
```

Comment:
Just thinking aloud. If I perform a small esearch that returns 2 articles. Is the only way to save them to database, via writing CSV?

Reply:
Right now, for MySQL, yes.  But the default option when saving an efetch is to delete the csv files along the way.